src/esp/gfx/GenericShader.cpp

```diff
@@ -160,8 +162,11 @@ GenericShader& GenericShader::bindTexture(Magnum::GL::Texture2D& texture) {
 
   texture.bind(TextureLayer);
 
+// TODO this is a hack and terrible! Properly set texSize for WebGL builds
```

Comment:
I guess having `bindTexture(Texture2D& texture, int size)` would be a solution, right? Unfortunately until ES 3.1 there's no way to query texture image sizes.